synapse/lib/layer.py

```diff
@@ -1204,6 +1224,99 @@
 
         self.onfini(self._onLayrFini)
 
+        # if we are a mirror, we upstream all our edits and
+        # wait for them to make it back down the pipe...
+        self.leader = None
+        self.leadtask = None
+        self.ismirror = layrinfo.get('mirror') is not None
+
+    @contextlib.contextmanager
+    def getIdenFutu(self, iden=None):
+
+        if iden is None:
+            iden = s_common.guid()
+
+        futu = self.loop.create_future()
+        self.futures[iden] = futu
+
+        yield iden, futu
+
+        self.futures.pop(iden, None)
+
+    async def getMirrorStatus(self):
+        # TODO plumb back to upstream on not self.core.isactive
+        retn = {'mirror': self.leader is not None}
+        if self.leader:
+            proxy = await self.leader.proxy()
+            retn['local'] = {'size': await self.getEditSize()}
+            retn['remote'] = {'size': await proxy.getEditSize()}
+        return retn
+
+    async def initLayerActive(self):
+
+        if self.leadtask is not None:
+            self.leadtask.cancel()
+
+        mirror = self.layrinfo.get('mirror')
+        if mirror is not None:
+            self.leader = await s_telepath.Client.anit(mirror)
+            self.leadtask = self.schedCoro(self._runMirrorLoop())
+
+    async def initLayerPassive(self):
+
+        if self.leadtask is not None:
+            self.leadtask.cancel()
+            self.leadtask = None
+
+        if self.leader is not None:
+            await self.leader.fini()
+            self.leader = None
+
+    async def getEditSize(self):
+        return self.nodeeditlog.size
+
+    async def _runMirrorLoop(self):
+
+        while not self.isfini:
+
+            try:
+
+                proxy = await self.leader.proxy()
+
+                leadoffs = await self._getLeadOffs()
+
+                async for offs, edits, meta in proxy.syncNodeEdits2(leadoffs + 1):
+
+                    iden = meta.get('task')
+                    futu = self.futures.pop(iden, None)
+
+                    meta['indx'] = offs
+
+                    try:
+                        item = await self.saveToNexs('edits', edits, meta)
```

Comment:
I haven't tested it, but I *think* if we got stuck with an edit that raised on save we would keep adding a new nexuslog entry every turn of this loop.